src/components/item/Item.js

```diff
@@ -1,85 +1,139 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useStore } from 'store/index';
 
-const Item = props => {
+const Item = ({ product }) => {
+    const { id, img, name, price, size: inventory } = product;
     const { dispatch, state } = useStore();
-    const { size } = state;
+    const [size, setSize] = useState();
+    const [quantity, setQuantity] = useState(1);
+    const [formErrors, setFormErrors] = useState({
+      currentlyInCart: false,
+      currentlyInWishlist: false,
+      size: false,
+      quantity: false,
+      outOfStock: false,
+      insufficientStock: false
+    });
 
-    const [numberOfItems, setNumberOfItems] = useState({ number: 1, id: null });
+    const handleAddToCartClick = () => {
+      const errors = {
+        quantity: quantity < 1,
+        size: !size,
+        currentlyInCart: !!state.cart.find(({ id: productId }) => id === productId),
+        outOfStock: inventory[size] < 1,
+        insufficientStock: quantity > inventory[size]
```

Comment:
Here we are declaring that the item form has insufficient stock if the user selected quantity exceeds the current inventory. Overall declarative-style state management is favored in react.